sdk/azidentity/auth_file_credential.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/url"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+type authFileAccesstoken struct {
+	ClientId                       string `json:"clientId"`
+	ClientSecret                   string `json:"clientSecret"`
+	SubscriptionId                 string `json:"subscriptionId"`
+	TenantId                       string `json:"tenantId"`
+	ActiveDirectoryEndpointUrl     string `json:"activeDirectoryEndpointUrl"`
+	ResourceManagerEndpointUrl     string `json:"resourceManagerEndpointUrl"`
+	ActiveDirectoryGraphResourceId string `json:"activeDirectoryGraphResourceId"`
+	SqlManagementEndpointUrl       string `json:"sqlManagementEndpointUrl"`
+	GalleryEndpointUrl             string `json:"galleryEndpointUrl"`
+	ManagementEndpointUrl          string `json:"managementEndpointUrl"`
+}
+
+// Enables authentication to Azure Active Directory using configuration information stored Azure SDK Auth File.
+type AuthFileCredential struct {
+	filePath   string
+	credential azcore.TokenCredential
+}
+
+// Creates an instance of the SdkAuthFileCredential class based on information in given SDK Auth file.
+func NewAuthFileCredential(filePath string) (*AuthFileCredential, error) {
+	if filePath == "" {
+		return nil, fmt.Errorf("The parameter 'filePath' must set a value")
+	}
+
+	return &AuthFileCredential{filePath: filePath}, nil
+}
+
+// Obtains a token from the Azure Active Directory service, using the specified client detailed specified in the SDK Auth file.
+func (c *AuthFileCredential) GetToken(ctx context.Context, opts azcore.TokenRequestOptions) (*azcore.AccessToken, error) {
+	_, err := c.ensureCredential()
+	if err != nil {
+		return nil, &AuthenticationFailedError{msg: "Error parsing SDK Auth File: " + err.Error()}
+	}
+
+	return c.credential.GetToken(ctx, opts)
+}
+
+// AuthenticationPolicy implements the azcore.Credential interface on AuthFileCredential.
+func (c *AuthFileCredential) AuthenticationPolicy(options azcore.AuthenticationPolicyOptions) azcore.Policy {
+	return newBearerTokenPolicy(c, options)
+}
+
+// Ensures that credential information is loaded from the SDK Auth file. This method should be called to initialize.
+func (c *AuthFileCredential) ensureCredential() (azcore.TokenCredential, error) {
+	if c.credential != nil {
+		return c.credential, nil
+	}
+
+	authData, err := c.parseCredentialsFile(c.filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	c.credential, err = c.buildCredentialForCredentialsFile(authData)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.credential, nil
+}
+
+// Parse credential file from local file path to byte.
+func (c *AuthFileCredential) parseCredentialsFile(filePath string) ([]byte, error) {
+	authData, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
I don't believe calling `log.Fatal(err)` is what we want here.  It will cause the process to exit  I think this should return `nil, err` if `error` is non `nil`.  If you agree, we could get rid of this method completely and just call `ioutil.ReadFile` in `ensureCredential`